nucleus/payara-modules/notification-core/src/main/java/fish/payara/nucleus/notification/NotifierHandler.java

```diff
@@ -146,17 +162,20 @@ public void accept(final PayaraNotification notification) {
         }
     }
 
-    @Override
-    public void run() {
-        final PayaraNotification notification = notificationQueue.peek();
+    private void run() {
+        assert isEnabled();
         try {
-            if (notification != null && isEnabled()) {
-                notifier.handleNotification(notification);
-                notificationQueue.remove();
+            final PayaraNotification notification = notificationQueue.take();
```

Comment:
take() is a blocking operation, and you are calling it every 5 milliseconds, so that will end up with a lot waiting on this if there aren't many notifications being sent, the opposite problem of what was wrong before. I think a better solution would be to flush the queue every 50 milliseconds or something like that.